pkg/security/probe/load_controller.go

```diff
@@ -108,35 +192,62 @@ func (lc *LoadController) discardNoisiestProcess() {
 	}
 
 	// update current total and remove biggest entry from cache
-	atomic.AddInt64(&lc.total, -int64(atomic.SwapUint64(maxCount, 0)))
+	oldMaxCount := atomic.SwapUint64(maxCount, 0)
+	atomic.AddInt64(&lc.eventsTotal, -int64(oldMaxCount))
 
 	if lc.statsdClient != nil {
 		// send load_controller.pids_discarder metric
 		tags := []string{
 			fmt.Sprintf("event_type:%s", maxKey.Event),
 		}
-		if err := lc.statsdClient.Count(MetricPrefix+".load_controller.pids_discarder", 1, tags, 1.0); err != nil {
+		if err := lc.statsdClient.Count(MetricLoadControllerPidDiscarder, 1, tags, 1.0); err != nil {
 			log.Warnf("couldn't send load_controller.pids_discarder metric: %v", err)
 			return
 		}
+
+		// fetch noisy process metadata
+		process := lc.probe.resolvers.ProcessResolver.Resolve(maxKey.Pid, maxKey.Cookie)
+
+		lc.probe.DispatchCustomEvent(
+			NewNoisyProcessEvent(
+				maxKey.Event,
+				oldMaxCount,
+				lc.EventsCountThreshold,
+				lc.ControllerPeriod,
+				time.Now().Add(lc.DiscarderTimeout),
```

Comment:
maybe you can call time.Now only once

Reply:
Let's use `event.ResolveEventTimestamp()` here as well.